dot/sync/bootstrap_syncer_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func newMockBlockStateForBootstrapSyncerTest(ctrl *gomock.Controller) BlockState {
+	mock := NewMockBlockState(ctrl)
+
+	//BestBlockHeader() (*types.Header, error)
+	mock.EXPECT().BestBlockHeader().Return(&types.Header{Number: 2}, nil)
+
+	mock.EXPECT().GetHighestFinalisedHeader().Return(&types.Header{Number: 1}, nil).AnyTimes()
+
+	// mock.EXPECT().GetHashByNumber(gomock.AssignableToTypeOf(uint(0))).DoAndReturn(func(
+	// 	number uint) (common.Hash, error) {
+	// 	return common.Hash{}, nil
+	// }).AnyTimes()
+
+	// mock.EXPECT().IsDescendantOf(gomock.AssignableToTypeOf(common.Hash{}),
+	// 	gomock.AssignableToTypeOf(common.Hash{})).Return(true, nil).AnyTimes()
+
+	// mock.EXPECT().GetHeader(gomock.AssignableToTypeOf(common.Hash{})).DoAndReturn(func(hash common.Hash) (*types.
+	// 	Header, error) {
+	// 	header := &types.Header{
+	// 		Number: uint(hash[0]),
+	// 	}
+	// 	return header, nil
+	// }).AnyTimes()
+
+	return mock
+}
+
+func Test_bootstrapSyncer_handleWorkerResult(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	tests := []struct {
```

Comment:
I think would be better to have a map[string]struct{...} then in the test loop you use the key as the name:

```go
for testName, tt := range tests {...}
```

Reply:
Updated.